lib/src/widgets/form_part.dart

```diff
@@ -211,7 +211,8 @@ class __FormPartState extends State<_FormPart> {
             else
               SizedBox(
                 height:
-                    loginTheme.spacingWithoutSocial ?? dynamicSize.height * 6,
+                  (loginTheme.spacingWithoutSocial!.compareTo(0) > 0
```

Comment:
Are we really sure that the "spacingWithoutSocial" field will not be null at this line of code? Did you check it? If you, can you indicate me the line which ensures that "spacingWithoutSocial" is not null.

Reply:
@bahricanyesil Good point, I didn't realize there was no null check previously. I added a null check, tested it and pushed the code. In my test, I specifically set the `spacingWithoutSocial` value to null. Will update the PR.

ps: I am relatively new to Flutter. Learning it by contributing to existing open source packages.

Reply:
No problem, I respectfully appreciate your effort to contribute something. All of us always learn in the road. I will point out some "nicer" code example for the part you changed.

However, can you explain me what problem you were experiencing and decided to make this change? What was the problem before and what is your expectation when "spacingWithoutSocial" is set to 0?

Thanks.

Reply:
When `spacingWithoutSocial` is set o 0 AND the social logins array is empty, there is not _padding_ between the form and title. I think that's the issue # 46 (bug). So, I updated the else conditions. The previous logic only handled `spacingWithoutSocial` being undefined / null. But, it did not handle `spacingWithoutSocial` being set to 0. Hence, my code change to handle both situations `spacingWithoutSocial` undefined/null or `spacingWithoutSocial` set to 0.

Reply:
Apart from the issue, for the suggestion part:

Why did you choose to use "compareTo" method here?

My suggestion:
You can some kinda "manipulate" the program instead of using long null check (with != null). I prefer this implementation:

`(loginTheme.spacingWithoutSocial ?? 0) > 0 ? loginTheme.spacingWithoutSocial : 100`

What do you think about this line, did it make sense to you?

Thanks for your effort.

Reply:
Also, in general, most UI elements use "some spacing" to have a nice transition. That said, I agree there could be someone that wants zero spacing. 

>(loginTheme.spacingWithoutSocial ?? 0) > 0 ? loginTheme.spacingWithoutSocial : dynamicSize.height * 2
That looks good. I don't have a specific reason to choose compareTo. 

Is it ok to make that change and push? I also see a discussion of EdgeInsets. Let me know.

Reply:
If you want to implement "EdgeInsets" as a challenge for you, you can work on it. If not, let me know so that we can merge this and then I will implement above it.

Reply:
I will first make the above change as a first pass. This will address the immediate issue, if you can merge.
Then, I want to take a shot at EdgeInsets so I can learn something new. Give me a day or two.

Reply:
Ok, I will merge this right now. However, just wait for my new update because it may fix the problem. I have worked on the feature that allows users to order the components as mentioned in [here](https://github.com/bahricanyesil/flutter-animated-login/issues/45#issue-1120741600). I will most probably merge it in 5-6 hours. After that, you can look at my changes and if you can see any problem, missing points can create a new branch and start to work on.

Reply:
Waiting for your change to resolve this conversation.

Reply:
Please take a look at my PR: #48 @mlakkapr